scripts/firedrake-install

```diff
@@ -1650,10 +1677,20 @@ else:
 
     with pipargs("--no-deps"):
         if args.rebuild or petsc_changed or petsc4py_changed:
-            clean("petsc4py/")
+            # If not honour_petsc_dir, we have a version of petsc that contains
+            # petsc4py in its source tree (old petsc must have been updated
+            # in the above).
+            # If honour_petsc_dir, we can get here (e.g., with --rebuild flag)
+            # without obtaining the latest petsc in the above:
+            # thus, we must check if the user installed petsc contains petsc4py.
+            if args.honour_petsc_dir:
+                if not os.path.exists(get_petsc4py_dir()):
+                    raise InstallError("petsc4py not found in your installation of petsc, "
+                                       "which indicates that your petsc is relatively old. "
+                                       "Please consider updating your petsc manually.")
```

Comment:
Multiple strings separated by whitespace are automatically concatenated by the parser. So this turns into one long line.

Reply:
Thanks Lawrence. So this is standard well-linted python?

Reply:
Thanks. I updated so that it would be easier to read.
I think `petsc` is constantly used in this script, and the name of the directory is also lower case, so maybe it's ok(?)

Reply:
A quick search shows that in most places where text is outputted to a user PETSc is used instead of petsc

Reply:
That's true. Fixed now. Thanks.